bigtable_automation/gcf/controller_trigger.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package gcf
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/pkg/errors"
+)
+
+const (
+	dcManifestPath = "/memfile/core_resolved_mcfs_memfile/core_resolved_mcfs.binarypb"
+)
+
+type PublishConfig struct {
+	// TopicName is the full PubSub topic name in the following format
+	// projects/<project id>/topics/<topic id>
+	FullTopicName string
```

Comment:
s/FullTopicName/TopicName, since there is no short topic name?

Reply:
thanks! (I didn't know TopicID was a thing).